test/test_constraints.py

```diff
@@ -69,3 +70,78 @@ def test_unknown_labels_not_in_chunks(num_chunks, chunk_size):
                               random_state=SEED)
 
   assert np.all(chunks[labels < 0] < 0)
+
+
+def test_generate_knntriplets():
```

Comment:
I think you should expand this test (manually or through `parametrize`) to check that everything goes well for several values of `k_genuine` and `k_impostor`, testing the normal case (the one you test already) but also the edge cases (when there is exactly the right amount of data points, and when one point is missing for either genuine or impostor neighbors).

Reply:
Agreed! I added a test on those edge cases and the result is compared to the result of a "naive" version of the knn generator.

Reply:
Why do you need to define a naive version? (which is then itself not tested...)
You could simply do everything in `test_generate_knntriplets` on the data you defined manually. Just like you did for the normal case, you can manually create the correct set of triplets for the edge cases and do the check

Reply:
Done!